### ISSUE TITLE
idris: enable gmp and ffi support

### ISSUE THREAD

**Issue by peti** [opensuse-haskell/configuration#8]
The spec file for [Idris](http://hackage.haskell.org/package/idris), for instance, should probably be created with `ffi` and `gmp` flags enabled.

**Comment by peti** [comment]
We cannot enable `ffi` because, apparently, Tumpleweed doesn't have a version of `libffi` that supports recent versions of `gcc`? All we have  is:

~~~
# zypper search libffi
Loading repository data...
Reading installed packages...

S | Name                    | Summary                                              | Type      
--+-------------------------+------------------------------------------------------+-----------
  | libffi-devel-gcc5       | Foreign Function Interface library development files | package   
  | libffi-devel-gcc5-32bit | Foreign Function Interface library development files | package   
  | libffi-gcc5             | The GNU C Compiler and Support Files                 | srcpackage
i | libffi4                 | Foreign Function Interface library                   | package   
i | libffi4-32bit           | Foreign Function Interface library                   | package   
  | uwsgi-libffi            | libffi Plugin for uWSGI                              | package   
~~~

and none of those seems correct for Idris.

**Comment by peti** [comment]
Enabling `gmp` requires `ffi`.

**Comment by trommler** [comment]
Did you try gmp-devel and libffi-devel as build requires? They work with ghc in Tumbleweed just fine.

**Comment by mimi1vx** [comment]
peti @trommler  is right .. ghc is build with libffi-devel

libffi-gcc5 is from gcc5 compiler, 
but libffi7 from gcc6.x is disabled in GCC6 build .. it needs ask rguenther@suse.com why .

**Comment by peti** [comment]
I tried `libffi-devel` before, but I got scared away when I saw that it resolves to `libffi-devel-gcc5` -- which doesn't seem right for a ghc6-based installation, but I only now realized that the library actually doesn't depend on the gcc version that built it, i.e. it seems fine to use that gcc5-built library in a gcc6-built application. So I gave this another shot now our `idris` package has the `gmp` and `ffi` Cabal flags enabled (in lts-7).
